### DISCUSSION TITLE
Can settings/profiles be programmatically set/imported

### DISCUSSION THREAD

**Discussion by AdreKiseque** [microsoft/terminal#18380] [Q&A] [answered]
I'm wondering if it's possible to set up my profiles and settings through a PowerShell script. It looks like the `settings.json` uses machine-unique values so I worry it wouldn't be possible to just import it directly.

**Comment by DHowett** [comment]
Absolutely! The `guid` isn't machine-specific, it's just there to make sure two profiles with the same name don't collide.

You can even generate a folder of JSON files containing only profiles and color schemes! We call them [fragment extensions](https://learn.microsoft.com/en-us/windows/terminal/json-fragment-extensions).

**Reply by AdreKiseque** [reply]
Ooh, thanks for the pointer. Looks like I had a misunderstanding of what the GUIDs in there represented...
Still, I'm not sure I quite fully understand. That is, I see there _is_ a way to do what I want, but I wasn't able to gleam quite how from the documentation.
I'd like to set up a full set of profiles/settings to import onto a new device. What's the best way to go about this? Can I bring over `settings.json` directly? Are these "fragments" a sufficient tool for this task?

**Reply by AdreKiseque** [reply]
@DHowett sorry, just wondering if you could clear things up a bit more?

**Reply by DHowett** [reply, marked as answer]
I'm sorry about the delay; clearly I do not have my inbox set up properly!

So, yes, you can _absolutely_ bring `settings.json` over to your new device. It only contains minimal machine-specific configuration, and you should be able to figure out where it messes up, if anywhere (mostly: if you have a WSL profile that exists on one machine but not the others.)
I missed that part of your original query!

You can also set up specifically *profiles* and *color schemes* in a generic format that you can easily move between machines.

For example, if you create `myprofiles.json` in `%LOCALAPPDATA%\Microsoft\Windows Terminal\Fragments\AdreKiseque`, you can put your own color schemes and profiles into it which will live outside your `settings.json` file:

```json
{
  "profiles": [
    {
      "commandline": "pwsh",
      "name": "PowerShell, but with some flavor",
      "colorScheme": "Dracula",
      "startingDirectory": "~"
    }
  ],
  "schemes": [
    {
      "name": "Dracula",
      "cursorColor": "#F8F8F2",
      "selectionBackground": "#44475A",
      "background": "#282A36",
      "foreground": "#F8F8F2",
      "black": "#21222C",
      "blue": "#BD93F9",
      "cyan": "#8BE9FD",
      "green": "#50FA7B",
      "purple": "#FF79C6",
      "red": "#FF5555",
      "white": "#F8F8F2",
      "yellow": "#F1FA8C",
      "brightBlack": "#6272A4",
      "brightBlue": "#D6ACFF",
      "brightCyan": "#A4FFFF",
      "brightGreen": "#69FF94",
      "brightPurple": "#FF92DF",
      "brightRed": "#FF6E6E",
      "brightWhite": "#FFFFFF",
      "brightYellow": "#FFFFA5"
    }
  ]
}
```

If you put that on every machine, in the same place, that profile will be available on all of them--regardless of the version of Terminal, and even if you destroy your personal settings file.

**Reply by AdreKiseque** [reply]
Wonderful, thanks!
